File: exo3.3.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>


/*Ecrire un programme qui saisit un nom de fichier texte au clavier et ouvre ce
fichier dans l'editeur emacs, dont le fichier executable se trouve a l'emplacement /usr/bin/emacs.*/
int main (int argc, char** argv) {

    if (argc != 2) {
        fprintf (stderr, "Veuillez le nom d'un fichier en argument.\n");
        exit (0);
    }

    char* fichier = argv[1];
    fprintf (stdout, "Ouverture de %s dans emacs ... \n", fichier);

    execv ("/usr/bin/emacs", "emacs", fichier, NULL);

    return 0;
}

File: exo3.3.py
#!/usr/bin/env python3
import sys
import os

# Programme qui ouvre un fichier dans l'éditeur emacs

def main():
    if len(sys.argv) != 2:
        print("Veuillez le nom d'un fichier en argument.", file=sys.stderr)
        sys.exit(0)
    
    fichier = sys.argv[1]
    print(f"Ouverture de {fichier} dans emacs ...")
    
    # Correction: execv prend deux arguments: chemin et liste d'arguments
    os.execv("/usr/bin/emacs", ["/usr/bin/emacs", fichier])

if __name__ == "__main__":
    main()